spec/features/idv/inherited_proofing/inherited_proofing_cancel_spec.rb

```diff
@@ -0,0 +1,257 @@
+require 'rails_helper'
+
+# Simulates a user (in this case, a VA inherited proofing-authorized user)
+# coming over to login.gov from a service provider, and hitting the
+# OpenidConnect::AuthorizationController#index action.
+def send_user_from_service_provider_to_login_gov_openid_connect(user)
+  expect(user).to_not be_nil
+  # NOTE: VA user.
+  visit_idp_from_oidc_va_with_ial2
+end
+
+def complete_idv_steps_up_to_inherited_proofing_get_started_step(user, expect_accessible: false)
+  unless current_path.match?(/inherited_proofing[?|\/].*get_started/)
+    complete_idv_steps_before_phone_step(user)
+    click_link t('links.cancel')
+    click_button t('idv.cancel.actions.start_over')
+    expect(page).to have_current_path("#{idv_inherited_proofing_path}/get_started")
+  end
+  expect(page).to be_axe_clean.according_to :section508, :"best-practice" if expect_accessible
+end
+
+def complete_idv_steps_up_to_inherited_proofing_how_verifying_step(user, expect_accessible: false)
+  complete_idv_steps_up_to_inherited_proofing_get_started_step user,
+                                                               expect_accessible: expect_accessible
+  unless current_path.match?(/inherited_proofing[?|\/].*agreement/)
+    click_on t('inherited_proofing.buttons.continue')
+  end
+end
+
+def complete_idv_steps_up_to_inherited_proofing_we_are_retrieving_step(user,
+                                                                       expect_accessible: false)
+  complete_idv_steps_up_to_inherited_proofing_how_verifying_step(
+    user,
+    expect_accessible: expect_accessible,
+  )
+  unless current_path.match?(/inherited_proofing[?|\/].*verify_wait/)
+    check t('inherited_proofing.instructions.consent', app_name: APP_NAME), allow_label_click: true
+    click_on t('inherited_proofing.buttons.continue')
+  end
+end
+
+def complete_idv_steps_up_to_inherited_proofing_verify_your_info_step(user,
+                                                                      expect_accessible: false)
+  complete_idv_steps_up_to_inherited_proofing_we_are_retrieving_step(
+    user,
+    expect_accessible: expect_accessible,
+  )
+end
+
+feature 'inherited proofing cancel process', :js do
+  include InheritedProofingHelper
+  include_context 'va_user_context'
+
+  before do
+    allow(IdentityConfig.store).to receive(:va_inherited_proofing_mock_enabled).and_return true
+    send_user_from_service_provider_to_login_gov_openid_connect user
+  end
+
+  let!(:user) { user_with_2fa }
+
+  context 'when JS is enabled', :js do
+    context 'from the "Get started verifying your identity" view, and clicking the "Cancel" link' do
+      before do
+        complete_idv_steps_up_to_inherited_proofing_get_started_step user
+      end
+
+      it 'should have current path equal to the Getting Started page' do
+        expect(page).to have_current_path(/inherited_proofing[?|\/].*get_started/)
```

Comment:
Could you explain the need for a regular expression here, as opposed to the URL helpers used elsewhere (e.g. `idv_inherited_proofing_cancel_path(step: :get_started)`)?

Reply:
Yes. Depending on what's controlling the redirecting, the paths are using query string or rails route path helpers; this helped me to focus on the test and not what was driving the redirecting - it's pretty safe.

Reply:
Could we use the [`ignore_query` option of `have_current_path`](https://www.rubydoc.info/gems/capybara/Capybara/SessionMatchers#assert_current_path-instance_method) instead, in that case?

My general concerns with a regular expression would be:

- Correctness
  - For example, this pattern would pass for a path of `/wrong/the_inherited_proofing/not_the_get_started_step/`
- Readability for future maintainers
- Hard-coding paths makes it difficult to change paths if we needed to in the future (where URL helpers help keep `routes.rb` as the single source of truth for the specific path detail)

Reply:
Argh, yea I know. However, this is (the query param part) used by the cancellations from IP flow steps, and the redirects using `idv_inherited_proofing_cancel` - we're forced to use query params so that path can be used to cancel from 1-n flow steps. IOW, there are no `idv_inherited_proofing_cancel_<flow step>_path` helpers to use. 

Perhaps I can use an or regexp that removes the .* which might make it more clear?

Reply:
I think the only way around the regex is to change the test to check a bool; bool set by checking current_path with an OR condition; would that get us past this?

```ruby
path_acceptable = current_path == 'xxx/get_started' || current_path == 'xxx?step=get_started'
expect(path_acceptable).to eq true
```

Reply:
Ah, I guess I should have checked more closely; I know that Zach made some changes, the changes he made probably eliminates the need, ty.

Reply:
+1 to Andrew's suggestion. That should work for whatever the URL we support is and will be less brittle and more readable than a comparison to a regex

Reply:
It does, just pushed it up.

Reply:
Done.